HelpSource/Reference/Functions.schelp

```diff
@@ -53,52 +71,82 @@ code::{ |x = 1| ... } :: link::#[1]::
 Pipe style, default value is an expression:
 code::{ |x = (10.rand)| ... } :: link::#[2]::
 
-If the last argument in the list is preceded by three dots (an ellipsis), then all the remaining arguments that were passed will be assigned to that variable as an link::Classes/Array::. Arguments must be separated by commas.
 
-examples:
+In general arguments may be initialized to literals or expressions, but in the case of Function:play or SynthDef:play, they may only be initialized to literals.
 code::
-{ arg a, b, c=3; } // is equivalent to:
+// this is okay:
 
-{ |a, b, c=3| }
+{ arg a = Array.geom(4, 100, 3); a * 4 }.value;
 
-{ arg x='stop', y, z=0; } // these args are initialised
+// this is not:
 
-{ arg a, b, c ... d; } // any arguments after the first 3 will be assigned to d as an Array
+{ arg freq = Array.geom(4, 100, 3); Mix(SinOsc.ar(freq, 0, 0.1)) }.play; // silence
+
+// but this is:
+{ arg freq =  #[ 100, 300, 900, 2700 ]; Mix(SinOsc.ar(freq, 0, 0.1)) }.play; // silence
 ::
 
-If you want all the arguments put in an Array
+See link::Reference/Literals:: for more information.
+
+
+
+section:: Variable Arguments
+
+Variable arguments are how function catch an unknown number of arguments or keyword arguments. 
+There are two possible variable argument, the first for positional arguments and second for keyword arguments.
+They are marked by an ellipsis after any normal arguments.
+
+The first variable argument catches all remaining arguments in an link::Classes/Array::
+
 code::
-arg ... z;
+f = { |a ... args| "a: %, args: %".format(a, args) };
+
+f.(1, 2, 3) // a: 1, args: [2, 3]
+f.(1) // a: 1, args: []
+f.(a: 1) // a: 1, args: []
+
+// catches all arguments
+g = { |...args| "args: %".format(args) };
+
+g.(1, 2, 3) // args: [1, 2, 3]
+g.() // args: []
 ::
 
-In general arguments may be initialized to literals or expressions, but in the case of Function:play or SynthDef:play, they may only be initialized to literals.
+The second variable argument catches all keyword arguments.
+It put them in an link::Classes/Array:: of key-value pairs.
+
 code::
-// this is okay:
+f = { |...args, kwargs| "args: %, kwargs: %".format(args, kwargs) };
 
-{ arg a = Array.geom(4, 100, 3); a * 4 }.value;
+f.(1, 2, 3) // args: [1, 2, 3], kwargs: []
 
-// this is not:
+f.(a: 10) // args: [], kwargs: [a, 10]
 
-{ arg freq = Array.geom(4, 100, 3); Mix(SinOsc.ar(freq, 0, 0.1)) }.play; // silence
+f.(1, 2, 3, meow: 10, woof: 20) // args: [1, 2, 3], kwargs: [meow, 10, woof, 20]
 
-// but this is:
-{ arg freq =  #[ 100, 300, 900, 2700 ]; Mix(SinOsc.ar(freq, 0, 0.1)) }.play; // silence
 ::
 
-See link::Reference/Literals:: for more information.
+The use of keyword variable arguments is often paired with link::Classes/Object#-performArgs::.
+
+You cannot reassign the value of the variable arguments even if they share the same name.
```

Comment:
How about this?
```
You cannot reassign the value of the variable arguments inside the keyword variable arguments, even if they share the same name.
```

Reply:
Yes, we need a better name for the keyword arguments in the function. A shorter one would be nice though. Maybe "inner keyword arguments"?